plugin/src/test/java/org/opensearch/ml/tools/ToolIntegrationWithLLMTest.java

```diff
@@ -63,9 +72,39 @@ public void stopMockLLM() {
     @After
     public void deleteModel() throws IOException {
         undeployModel(modelId);
+        waitModelUndeployed(modelId);
         deleteModel(client(), modelId, null);
     }
 
+    @SneakyThrows
+    private void waitModelUndeployed(String modelId) {
+        Predicate<Response> condition = response -> {
+            try {
+                Map<String, Object> responseInMap = parseResponseToMap(response);
+                MLModelState state = MLModelState.from(responseInMap.get(MLModel.MODEL_STATE_FIELD).toString());
+                return Set.of(MLModelState.UNDEPLOYED, MLModelState.DEPLOY_FAILED).contains(state);
+            } catch (Exception e) {
+                return false;
+            }
+        };
+        waitResponseMeetingCondition("GET", "/_plugins/_ml/models/" + modelId, null, condition);
+    }
+
+    @SneakyThrows
+    protected Response waitResponseMeetingCondition(String method, String endpoint, String jsonEntity, Predicate<Response> condition) {
+        for (int i = 0; i < MAX_TASK_RESULT_QUERY_TIME_IN_SECOND; i++) {
```

Comment:
The worst case this line will run `MAX_TASK_RESULT_QUERY_TIME_IN_SECOND` times, it will be 300 ? For every retry, it will sleep one second, so it will wait for 300 seconds, that's a long time. Do we really need so long time ? We should fail quickly with a reasonable time

Reply:
yeah, reduce to total 30s that should be enough for model to be undeployed.